packages/material-ui-lab/src/DataGrid/DataGrid.d.ts

```diff
@@ -21,14 +21,24 @@ export type SortingType = Array<{
 
 export interface DataProviderGetListParams {
   sorting: SortingType;
+  pagination: PagingOptions;
 }
 
 export interface DataProviderType {
   getList: (params: DataProviderGetListParams) => Promise<any[]>;
+  loadMoreRows?: (paginationKey: string) => Promise<string>;
+}
+
+export type PagingOptions = {
+  pagination: boolean;
+  paginationPage: number;
+  paginationPageSize: 10 | 25 | 50 | 100 | 250 | 500 | number;
+  paginationRowsPerPageOptions: number[];
+  paginationKey: string;
 }
 
 export interface DataGridProps
-  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, DataGridClassKey> {
+  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, DataGridClassKey>, Partial<PagingOptions> {
```

Comment:
We need to give a description to each prop

```suggestion
  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, DataGridClassKey> {
```